## Business Problem

As per the business problem, I am looking to determine the safest aircraft that The Company can purchase

## The Data

The data is a csv file from the National Transport Safety Board and includes aviation accident data from 1962 to 2023 about civil aviation accidents and selected incidents in the UNited Sates and international waters.

# 1. Exploratory Data Analysis

In this section I will look at the dataset and find key features and insights.

I will first import the relevant pandas package with its alias.

## a) Import Pandas and read the csv file

In [2]:
import pandas as pd

aviation_df = pd.read_csv('Aviation_Data.csv')
aviation_df.head()

/tmp/ipykernel_73711/1615477173.py:3: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  aviation_df = pd.read_csv('Aviation_Data.csv')


,Event.Id,Investigation.Type,Accident.Number,Event.Date,Location,Country,Latitude,Longitude,Airport.Code,Airport.Name,...,Purpose.of.flight,Air.carrier,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight,Report.Status,Publication.Date
0,20001218X45444,Accident,SEA87LA080,1948-10-24,"MOOSE CREEK, ID",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,UNK,Cruise,Probable Cause,NaN
1,20001218X45447,Accident,LAX94LA336,1962-07-19,"BRIDGEPORT, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,4.0,0.0,0.0,0.0,UNK,Unknown,Probable Cause,19-09-1996
2,20061025X01555,Accident,NYC07LA005,1974-08-30,"Saltville, VA",United States,36.922223,-81.878056,NaN,NaN,...,Personal,NaN,3.0,NaN,NaN,NaN,IMC,Cruise,Probable Cause,26-02-2007
3,20001218X45448,Accident,LAX96LA321,1977-06-19,"EUREKA, CA",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,2.0,0.0,0.0,0.0,IMC,Cruise,Probable Cause,12-09-2000
4,20041105X01764,Accident,CHI79FA064,1979-08-02,"Canton, OH",United States,NaN,NaN,NaN,NaN,...,Personal,NaN,1.0,2.0,NaN,0.0,VMC,Approach,Probable Cause,16-04-1980


## b) EDA

Next, I will look at how many rows and columns there are.

In [4]:
aviation_df.shape

(90348, 31)

From this I can see that there are 90,348 records with 31 columns. Taking a look at the columns:

In [5]:
aviation_df.columns

Index(['Event.Id', 'Investigation.Type', 'Accident.Number', 'Event.Date',
       'Location', 'Country', 'Latitude', 'Longitude', 'Airport.Code',
       'Airport.Name', 'Injury.Severity', 'Aircraft.damage',
       'Aircraft.Category', 'Registration.Number', 'Make', 'Model',
       'Amateur.Built', 'Number.of.Engines', 'Engine.Type', 'FAR.Description',
       'Schedule', 'Purpose.of.flight', 'Air.carrier', 'Total.Fatal.Injuries',
       'Total.Serious.Injuries', 'Total.Minor.Injuries', 'Total.Uninjured',
       'Weather.Condition', 'Broad.phase.of.flight', 'Report.Status',
       'Publication.Date'],
      dtype='object')

Since my main priority is identifying the safest aircraft for The Company I will focus on these 10 columns:
* 'Investigation.Type'
* 'Injury.Severity'
* 'Aircraft.damage'
* 'Registration.Number'
* 'Total.Fatal.Injuries'
* 'Total.Serious.Injuries'
* 'Total.Minor.Injuries'
* 'Total.Uninjured'
* 'Weather.Condition'
* 'Broad.phase.of.flight'

I will continue to explore the dataset.

In [6]:
aviation_df.describe

<bound method NDFrame.describe of              Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  U

In [7]:
aviation_df.info

<bound method DataFrame.info of              Event.Id Investigation.Type Accident.Number  Event.Date  \
0      20001218X45444           Accident      SEA87LA080  1948-10-24   
1      20001218X45447           Accident      LAX94LA336  1962-07-19   
2      20061025X01555           Accident      NYC07LA005  1974-08-30   
3      20001218X45448           Accident      LAX96LA321  1977-06-19   
4      20041105X01764           Accident      CHI79FA064  1979-08-02   
...               ...                ...             ...         ...   
90343  20221227106491           Accident      ERA23LA093  2022-12-26   
90344  20221227106494           Accident      ERA23LA095  2022-12-26   
90345  20221227106497           Accident      WPR23LA075  2022-12-26   
90346  20221227106498           Accident      WPR23LA076  2022-12-26   
90347  20221230106513           Accident      ERA23LA097  2022-12-29   

              Location        Country   Latitude  Longitude Airport.Code  \
0      MOOSE CREEK, ID  Uni

I am focusing only on accidents. I will examine the investigation type column to see what values it has

In [8]:
aviation_df['Investigation.Type'].value_counts

<bound method IndexOpsMixin.value_counts of 0        Accident
1        Accident
2        Accident
3        Accident
4        Accident
           ...   
90343    Accident
90344    Accident
90345    Accident
90346    Accident
90347    Accident
Name: Investigation.Type, Length: 90348, dtype: object>

All of the investigation types are accidents.

Let me now move on to my main columns and examine their values:

In [36]:
# I started by storing the main columns in a list
main_columns = ['Investigation.Type', 'Injury.Severity', 
                'Aircraft.damage', 'Registration.Number', 
                'Total.Fatal.Injuries', 'Total.Serious.Injuries', 
                'Total.Minor.Injuries', 'Total.Uninjured', 
                'Weather.Condition', 'Broad.phase.of.flight']

null_value_counts = [] # I then initialized a list that will store the corresponding count of the null values for each of the main columns

for column in main_columns:
    null_values_series = aviation_df[column].isnull().value_counts() # Get the count of null and filled values for 'column'
    shape = list(null_values_series.shape)
    if shape[0] > 1: # This essentially checks to see if the column has null values
        null_count = null_values_series.loc[True] # Get the count of the null values
        null_value_counts.append(int(null_count)) # Store the count of null values for 'column' in the list of null value counts
    else:
        null_value_counts.append(0)

null_values_dict = dict(zip(main_columns, null_value_counts)) # Create a dictionary mapping every column to its null value count
null_values_dict

{'Investigation.Type': 0,
 'Injury.Severity': 2459,
 'Aircraft.damage': 4653,
 'Registration.Number': 2841,
 'Total.Fatal.Injuries': 12860,
 'Total.Serious.Injuries': 13969,
 'Total.Minor.Injuries': 13392,
 'Total.Uninjured': 7371,
 'Weather.Condition': 5951,
 'Broad.phase.of.flight': 28624}

I will create a subset from the main dataframe containing these main columns and drop the null values and see what I get.

In [37]:
aviation_df[main_columns].dropna()

,Investigation.Type,Injury.Severity,Aircraft.damage,Registration.Number,Total.Fatal.Injuries,Total.Serious.Injuries,Total.Minor.Injuries,Total.Uninjured,Weather.Condition,Broad.phase.of.flight
0,Accident,Fatal(2),Destroyed,NC6404,2.0,0.0,0.0,0.0,UNK,Cruise
1,Accident,Fatal(4),Destroyed,N5069P,4.0,0.0,0.0,0.0,UNK,Unknown
3,Accident,Fatal(2),Destroyed,N1168J,2.0,0.0,0.0,0.0,IMC,Cruise
6,Accident,Fatal(4),Destroyed,N4988E,4.0,0.0,0.0,0.0,IMC,Unknown
7,Accident,Non-Fatal,Substantial,N2482N,0.0,0.0,0.0,2.0,VMC,Takeoff
...,...,...,...,...,...,...,...,...,...,...
63893,Accident,Non-Fatal,Substantial,N5875Q,0.0,0.0,2.0,0.0,VMC,Takeoff
63896,Incident,Incident,Minor,N47BC,0.0,0.0,0.0,4.0,VMC,Climb
63900,Accident,Non-Fatal,Substantial,N365SX,0.0,1.0,0.0,0.0,VMC,Cruise
63906,Accident,Fatal(2),Destroyed,N136DG,2.0,0.0,0.0,0.0,VMC,Maneuvering
